### PR TITLE
[ENH] Series toset() functionality, fixes #570

### LINKED ISSUE #570: [ENH] Series toset() functionality
# Brief Description

I would like to propose toset() functionality similar to [tolist()](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.tolist.html). 
Basically it will be a function call to tolist and then conversion to set.
Note: if the collection has no hashable member it will raise an exception.

# Example API

Please modify the example API below to illustrate your proposed API, and then delete this sentence.

```python
# convert series to a set
df['col1'].toset()

### PR DESCRIPTION
**Pull request by eyaltrabelsi**
# PR Description

Please describe the changes proposed in the pull request: 

- Added Series toset() functionality

**This PR resolves #570**

# PR Checklist

Please ensure that you have done the following:

1. [x] PR in from a fork off your branch. Do not PR from `<your_username>`:master, but rather from `<your_username>`:<branch_name>.

2. [x] If you're not on the contributors list, add yourself to `AUTHORS.rst`.

3. [x] Add a line to `CHANGELOG.rst` under the latest version header (i.e. the one that is "on deck") describing the contribution.
    - Do use some discretion here; if there are multiple PRs that are related, keep them in a single line.

## Quick Check

To do a very quick check that everything is correct, follow these steps below:

- [x] Run the command `make check` from pyjanitor's top-level directory. This will automatically run:
    - black formatting
    - pycodestyle checking
    - running the test suite
    - docs build
    
Once done, please check off the check-box above.
    
If `make check` does not work for you, you can execute the commands listed in the Makefile individually.

## Code Changes

If you are adding code changes, please ensure the following:

- [x] Ensure that you have added tests.
- [x] Run all tests (`$ pytest .`) locally on your machine.
    - [x] Check to ensure that test coverage covers the lines of code that you have added.
    - [ ] Ensure that all tests pass.

## Documentation Changes

If you are adding documentation changes, please ensure the following:

- [x] Build the docs locally.
- [x] View the docs to check that it renders correctly.

# Relevant Reviewers

Please tag maintainers to review.

- @ericmjl

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -4,6 +4,7 @@ new version (on deck)
 
 v0.18.2
 =======
+- [ENH] Series toset() functionality #570
 - [INF] Make requirements.txt smaller
```
Comment (ericmjl) [inline CHANGELOG.rst]: @eyaltrabelsi I think both lines 7 and 8 should have been put inside the "new version (on deck)" section, as 0.18.2 has already been released.

Can you move them please?

Comment (eyaltra) [reply CHANGELOG.rst]: @ericmjl I will move them, also i was creating the docs locally and couldn't find toset although i think i have added  all changes required

**Comment by ericmjl** [comment]
Looks like everything is set. I removed the WIP - @eyaltrabelsi, can you confirm?

**Comment by eyaltra** [comment]
@ericmjl That's a nice wordplay :) and it's ready :)

**Pull request merged**